src/mono/netcore/Makefile

```diff
@@ -0,0 +1,35 @@
+DOTNET=../../../.dotnet/dotnet
+TESTHOST_PATH=../../../artifacts/bin/testhost/
+
+# let's check ../../../artifacts/bin/testhost/ for available configurations
+# it's expected to see there only a single config, e.g. "netcoreapp5.0-OSX-Release-x64"
+# otherwise you have to set these variables for desired config
+ifeq ($(words $(wildcard $(TESTHOST_PATH)/*)), 1)
+	TESTHOST_CFG := $(subst -, ,$(notdir $(wildcard $(TESTHOST_PATH)/*)))
+	CONFIGURATION := $(word 3,$(TESTHOST_CFG))
+endif
```

Comment:
can we just let the defaults from mono.proj be used instead of this guesswork? why do we need to pass `/p:Configuration` all the time to mono.proj?

Reply:
@akoeplinger because I don't know with what parameters the testhost was built (and e.g. if there are more than one). `Configuration` is hard-coded to Debug always now https://github.com/dotnet/runtime/blob/master/src/mono/Directory.Build.props#L16

Reply:
I think it's fair to require the dev passing the same configuration as what was used for the overall build.

Reply:
@akoeplinger Makefile extracts the configuration used for the overall build and pass it to msbuild targets. Not sure I can do it in the mono.proj -- what if Configuration was set by someone else and I'll just override it always?

Reply:
I'd only set the `/p:Configuration` if the CONFIGURATION variable is set.

Reply:
@akoeplinger do you mean like this https://github.com/dotnet/runtime/pull/2393/commits/5fb59f6a1e0f600467017ef9cc33b0739e03c3cb ?

Reply:
More like this:

```diff
-# let's check ../../../artifacts/bin/testhost/ for available configurations
-# it's expected to see there only a single config, e.g. "netcoreapp5.0-OSX-Release-x64"
-# otherwise you have to set these variables for desired config
-ifeq ($(words $(wildcard $(TESTHOST_PATH)/*)), 1)
-	TESTHOST_CFG := $(subst -, ,$(notdir $(wildcard $(TESTHOST_PATH)/*)))
-	CONFIGURATION := $(word 3,$(TESTHOST_CFG))
-	MONO_PROPS := /p:Configuration=$(CONFIGURATION)
-endif
+ifneq($(CONFIGURATION),)
+	MONO_PROPS := /p:Configuration=$(CONFIGURATION)
+endif
```

Reply:
@akoeplinger it's not super convenient if you build the whole repo in Release config and as you remember Configuration is always `Debug` - all rules will require you to set config every time. The current code handles it.

Reply:
that's true but the same happens elsewhere. I'm not too obsessed with this though so I'd be fine with keeping this given that it's hopefully only temporary :)

Reply:
it is temporarily. I am just pretty sure it will be the first question "why should I have to set Configuration every time in all makefile rules" :)